### ISSUE TITLE
 help request, it appears my request is malformed; error ansi x9.42 "invalid hash algorithm supplied" 

### ISSUE THREAD

**Issue by mark-simonton** [usnistgov/ACVP#1461]
I am attempting to download vectors for _ansi x9.42_.  I am using the Identity of my alter-ego, because my account is in transition.
my request file is mostly a copy of the sample in **ACVP ANS x9.42 Key Derivation Function JSON Specification**, with some modifications to be more like request files that work for other algorithms.
this is the most recent iteration of my request file:
```
[
    {
      "acvVersion": "1.0"
    },
    {
        "isSample": false,
        "algorithms": [
        {
        "algorithm": "kdf-components",
        "mode": "ansix9.42",
        "revision": "1.0",
        "prereqVals": [
            {
                "algorithm": "SHA",
                "valValue": "same"
            }
        ],
        "kdfType": [
                "concatenation"
        ],
        "keyLen": [
                256
        ],
        "otherInfoLen": [
                512
        ],
        "suppInfoLen": [
                256
        ],
        "zzLen": [{
                "min":8,
                "max": 1024,
                "increment": 8
        }],
        "hashAlg": [
                "sha2-256"
        ]
        }
        ]
    }
]

```

here is the session log from my side.
Using the following parameters:

```
    ACV_SERVER:     demo.acvts.nist.gov
    ACV_PORT:       443
    ACV_URI_PREFIX: /acvp/v1/
    ACV_CA_FILE:    c:\vp\certs\mozzila_trust_anchors.pem
    ACV_CERT_FILE:  c:\vp\certs\Microsoft_Mark_Simonton_Demo.cer
    ACV_KEY_FILE:   c:\vp\certs\Microsft_Mark_simonton_Demo.key

***ACVP [STATUS][acvp_login:2723]--> Logging in...
***ACVP [STATUS][acvp_login:2743]--> Login successful
***ACVP [STATUS][acvp_register:2124]--> Reading capabilities registration file...
***ACVP [STATUS][acvp_register:2146]--> Sending registration of capabilities...
***ACVP [ERR][log_network_status:1247]--> 400 error received from server. Message:
***ACVP [ERR][log_network_status:1248]--> [
  {
    "acvVersion": "1.0"
  },
  {
    "error": "Validation error(s) on JSON payload.",
    "context": [
      "kdf-components-ansix9.42-1.0: Invalid Hash Alg supplied: sha2-256, intersect : 0, supplied: 1"
    ]
  }
]
***ACVP [ERR][acvp_register:2158]--> Failed to send registration
***ACVP [ERR][acvp_run:3488]--> Failed to register with ACVP server

```

thank you for your attention.
Mark Simonton

**Comment by livebe01** [comment]
Hi Mark, I believe you're running into a casing issue. Try using "SHA2-256" vs "sha2-256".

**Comment by mark-simonton** [comment]
new information for my request file I added json objects like 
```
		{
		"algorithm": "SHA2-256",
		"revision": "1.0",
		"messageLength": [
			{
			"min": 0,
			"max": 65536,
			"increment": 8
			}
		]
		}
```
before my request for ansix9.42  and I did receive a test vector set!!!

now to update our test harness to unserstand the input and produce a response file.

Also, a coworker, spliced the sample capabilities into our symcrypt capabilities and that worked also, since all the prerequisite SHA tests exist in that capability file for other tests.

thanks for the attention, sometimes you just need to write it out (and read it) to get a new perspective on the problem.
